### PR TITLE
Instantly show reward board updates

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Discussion](https://discord.com/channels/894960387743698944/1215442361832701952)

### CODE REVIEW

File: components/rewards/hooks/useRewardBlocks.tsx [hunk 1]
```diff
@@ -41,7 +44,19 @@ export function RewardBlocksProvider({ children }: { children: ReactNode }) {
   const { trigger: updateRewardBlocks } = useUpdateRewardBlocks(space?.id || '');
   const { trigger: deleteRewardBlocks } = useDeleteRewardBlocks(space?.id || '');
   const { showMessage } = useSnackbar();
-  const { boardBlock: rewardsBoardBlock, createProperty, deleteProperty, updateProperty } = useRewardsBoard();
+  const { createProperty, deleteProperty, updateProperty } = useRewardsBoard();
```
Comment (mattcasey) [inline components/rewards/hooks/useRewardBlocks.tsx:47]: why isnt the fix inside of useRewardsBoard?

Comment (Devorein) [reply components/rewards/hooks/useRewardBlocks.tsx:47]: Good point. Updated with ws

File: components/rewards/hooks/useRewardsBoard.ts [hunk 2]
```diff
@@ -32,6 +35,42 @@ export function useRewardsBoard() {
...
+            return [
+              {
+                ...existingRewardBoardBlock,
+                fields: Object.assign(existingRewardBoardBlock.fields, updatedRewardBoardBlock.fields)
+              }
+            ];
+          }
+
+          return rewardBlocks;
+        });
```
Comment (mattcasey) [inline components/rewards/hooks/useRewardsBoard.ts:60]: suggestion: no need to revaliadte, or else just revalidate and dont bother updating the cache. it's such a fast query

Comment (Devorein) [reply components/rewards/hooks/useRewardsBoard.ts]: Fair point. Updated

**Review by mattcasey — APPROVED** [review]

**Pull request merged**